sqlglot/expressions.py

```diff
@@ -1931,6 +1931,10 @@ class EngineProperty(Property):
     arg_types = {"this": True}
 
 
+class HeapProperty(Property):
+    arg_types = {}
```

Comment:
@bayees can you make sure that `HEAP` is a valid table option for T-SQL? I didn't find it in the documentation: https://learn.microsoft.com/en-us/sql/t-sql/statements/create-table-transact-sql?view=sql-server-ver16

Reply:
Nvm I just checked the link you provided in the issue: https://learn.microsoft.com/en-us/azure/synapse-analytics/sql-data-warehouse/sql-data-warehouse-develop-ctas